### ISSUE TITLE
Multi Store Product Update does not save data across all stores Magento 2.1.7

### ISSUE THREAD

**Issue by toddrobards** [magento/magento2#10400]
### Preconditions

1. Magento 2.1.7
2. MYSQL 5.6.36 MySQL Community Server (GPL) - Protocol version 10
3. PHP 5.6

### Steps to reproduce

1. Magento Installation has 2 websites / 3 stores
2. Through Admin panel
3. Add a product into Catalog with all stores selected
4. Reindex, Clear Cache, Clear Browser History --> Front end 3 websites display correctly
5. Update this product with new data (URL Key in tests below)
6. Reindex, Clear Cache, Clear Browser History --> 
  a. First website ("Product in Websites" / "Websites" list)  front end displays correctly as per update
  b. Other 2 websites selected in list do not update their data and displays old data
7. Sometimes, we can update the data by entering the new data, deselecting the other 2 stores, saving, then re selecting the 2 stores and re-saving. This has worked for other product updates, but now will not work for this URL Key error.

### Expected result

1. All websites should display the same product content as the updated product and updated website.

### Actual result

1. In the below screenshots, please look at the url which has been changed by update as above across 3 stores from "/two-ants-waist-extension-strap" to "/ctextbk". Default store is https://holstershop.com.au which is correct, but https://twoants.com.au is 404.

![screenshot 2017-08-02 14 34 19](https://user-images.githubusercontent.com/9101230/28857879-db461cae-778f-11e7-9406-515ddeb3310e.png)
![screenshot 2017-08-02 14 33 58](https://user-images.githubusercontent.com/9101230/28857883-e44f72d2-778f-11e7-9834-99a6a4398d1c.png)
![screenshot 2017-08-02 14 33 50](https://user-images.githubusercontent.com/9101230/28857888-eac3841e-778f-11e7-8d37-b3a1f5ff0edf.png)

**Comment by deebee92** [comment]
Hi @toddrobards, this sounds a bit like an issue we had not too long back, what we found was the scope where the data is saved to in the configuration was being overlooked for some reason so what we found to resolve it was go to stores > configuration > catalog > catalog and click to expand price then, select the relevant option either global/website (just select the one that is already ticked) and re-save the config.

You then need to clear the config cache again and go to make the product changes again. If this doesn't resolve your issue then apologies! I just thought I'd share this 😄

**Comment by toddrobards** [comment]
Hi @deebee92 , thanks for the info but unfortunately this didn't work for me.

**Comment by rukrlf** [comment]
Thanks @deebee92 it worked! 
I also faced a similar scenario. we have multiple websites in a single instance and did an upgrade recently from 2.0.15 to 2.1.7. Even when I saved the website level price field with a different price, it was showing old price, but when I change the scope to "All store view", it shows the price I enter. But since the website scope had the previous value, frontend was showing previous price. Our Australian shop client was complaining, she can't save price, when I checked, the price field of catalog->product was showing as global scope. I did what @deebee92 had mentioned, which is just re-saving Store->Config->Catalog->Catalog->Price->Catalog Price Scope. Keeping it as website, and catalog->product price field was showing website scope and after saving, it started appearing in frontend.
